cmd/containerboot/serve.go

```diff
@@ -59,24 +61,62 @@ func watchServeConfigChanges(ctx context.Context, path string, cdChanged <-chan
 			// k8s handles these mounts. So just re-read the file and apply it
 			// if it's changed.
 		}
-		if certDomain == "" {
-			continue
-		}
 		sc, err := readServeConfig(path, certDomain)
 		if err != nil {
-			log.Fatalf("failed to read serve config: %v", err)
+			log.Fatalf("serve proxy: failed to read serve config: %v", err)
 		}
 		if prevServeConfig != nil && reflect.DeepEqual(sc, prevServeConfig) {
 			continue
 		}
-		log.Printf("Applying serve config")
-		if err := lc.SetServeConfig(ctx, sc); err != nil {
-			log.Fatalf("failed to set serve config: %v", err)
+		validateHTTPSServe(certDomain, sc)
+		if err := updateServeConfig(ctx, sc, certDomain, kc, lc); err != nil {
+			log.Fatalf("serve proxy: error updating serve config: %v", err)
 		}
 		prevServeConfig = sc
 	}
 }
 
+func certDomainFromNetmap(nm *netmap.NetworkMap) string {
+	if len(nm.DNS.CertDomains) == 0 {
+		return ""
+	}
+	return nm.DNS.CertDomains[0]
+}
+
+func updateServeConfig(ctx context.Context, sc *ipn.ServeConfig, certDomain string, kc *kubeClient, lc *tailscale.LocalClient) error {
+	defer func() {
+		if err := kc.storeHTTPSEndpoint(ctx, certDomain); err != nil {
+			log.Printf("[unexpected]: serve proxy: error storing HTTPS endpoint: %v", err)
```

Comment:
I can see why we might not want to fatal out for this failure, but it does make me a bit apprehensive about a proliferation of invalid states we could get into when we hit combinations of "soft" failures. Did you see this error at all while developing the feature? I'd hope we can make these calls resilient enough to treat them as real errors.

Reply:
> proliferation of invalid states we could get into when we hit combinations of "soft" failures

Yeah I agree, I was worried about the same. I've changed this to fatal out. I've not oberved this error while testing